ladybug/wea.py

```diff
@@ -753,89 +900,80 @@ def estimate_illuminance_components(self, dew_point):
             'direct_normal_irradiance': self.direct_normal_irradiance.values,
             'diffuse_horizontal_irradiance': self.diffuse_horizontal_irradiance.values,
             'timestep': self.timestep,
             'is_leap_year': self.is_leap_year
         }
+        if not self.is_annual:
+            dts = self.direct_normal_irradiance.datetimes
+            base['datetimes'] = [dat.to_array() for dat in dts]
+        return base
 
-    def write(self, file_path, hoys=None, write_hours=False):
-        """Write the wea file.
+    def write(self, file_path, write_hours=False):
+        """Write the Wea object to a .wea file.
 
-        WEA carries irradiance values from epw and is what gendaymtx uses to
-        generate the sky.
+        Args:
+            file_path: Text string for the path to where the .wea file should be written.
+            write_hours: Boolean to note whether a .hrs file should be written
+                next to the .wea file, which lists the hours of the year (hoys)
+                contained within the .wea file.
         """
+        # prepare input file path and make header
         if not file_path.lower().endswith('.wea'):
             file_path += '.wea'
-        is_leap_year = self.is_leap_year
-        # generate hoys in wea file based on timestep
-        full_wea = False
-        if not hoys:
-            hoys = self.hoys
-            full_wea = True
-
-        # write header
         lines = [self.header]
-        if full_wea:
-            # there is no user input for hoys, write it for all the hours
-            for dir_rad, dif_rad, dt in zip(self.direct_normal_irradiance,
-                                            self.diffuse_horizontal_irradiance,
-                                            self.datetimes):
-                line = "%d %d %.3f %d %d\n" \
-                    % (dt.month, dt.day, dt.float_hour, dir_rad, dif_rad)
-                lines.append(line)
-        else:
-            # output wea based on user request
-            for hoy in hoys:
-                try:
-                    dir_rad, dif_rad = self.get_irradiance_value_for_hoy(hoy)
-                except IndexError:
-                    warnings.warn('Wea data for {} is not available!'.format(dt))
-                    continue
-
-                dt = DateTime.from_hoy(hoy, is_leap_year)
-                dt = dt.add_minute(30) if self.timestep == 1 else dt
-                line = "%d %d %.3f %d %d\n" \
-                    % (dt.month, dt.day, dt.float_hour, dir_rad, dif_rad)
-
-                lines.append(line)
+
+        # write values and datetimes
+        for dir_rad, dif_rad, dt in zip(self.direct_normal_irradiance,
+                                        self.diffuse_horizontal_irradiance,
+                                        self.datetimes):
+            line = "%d %d %.3f %d %d\n" \
+                % (dt.month, dt.day, dt.float_hour, dir_rad, dif_rad)
+            lines.append(line)
```

Comment:
This is much cleaner than what it used to be but I wonder how are you handling the 30 minutes shift for datetime? I see that the test file that is created looks correct but couldn't see where do you make the change in write method.

Reply:
The change for the datetimes happens in the datetimes property of the Wea:
https://github.com/ladybug-tools/ladybug/blob/master/ladybug/wea.py#L475-L482
You can see from where that code is that is how we were doing it in the past and I think it's a good solution. Admittedly, it means the datetimes property for the Wea might not be exactly as the datetimes are for the individual data collections. But this seems appropriate given that the Wea is following a different time standard from the rest of ladybug.

Reply:
Thank you for the clarification. This means that the file will be identical to the output of epw2wea command which is all we really need.